product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -132,4 +132,13 @@ def _select_customerinfo(
             .sorted(lambda s: (s.sequence, s.min_qty, s.price, s.id))
         )
         res_1 = res.sorted("product_tmpl_id")[:1]
-        return res_1
+        if res_1 or not partner.parent_id:
+            return res_1
+        else:
+            return self._select_customerinfo(
+                partner=partner.parent_id,
```

Comment:
Maybe can you use partner.commercial_partner_id ??

Reply:
@sergio-teruel thanks for your comment, sorry I was not able to respond before this was merged. Considering how the commercial_partner_id field is computed this would not do a fallback to the parent if the sub-contact in question is a company:
```python
            if partner.is_company or not partner.parent_id:
                partner.commercial_partner_id = partner
            else:
                partner.commercial_partner_id = partner.parent_id.commercial_partner_id
```
Maybe if you feel this would be the preferred behaviour you want to make a separate PR for this?